uktrade/data-hub-leeloo
# Docker compose rework

### Description of change

This reworks docker-compose usage slightly and improves the documentation.
- The key backward incompatible change (for docker users) is moving `docker-compose.env` to a `sample.env` file and requiring users to create a concrete `.env` file from this.  This copies the workflow from the data hub frontend project - increasing consistency between our docker approaches.
- The README has been modified to add an optional end step for setting up a local frontend app under docker-compose.  This delegates instructions to that repository.
- Additionally, the default environment variables for docker-compose have been modified to assume that `mocksso` is used as the SSO provider.

This PR should be reviewed in conjunction with the companion frontend PR: https://github.com/uktrade/data-hub-frontend/pull/1811

#### How to test

Please run through the instructions to bring up a docker-compose project locally.  
Then run through the instructions in the frontend PR (and by extension the mock-sso instructions).  
At the end of this, you should have local copies of leeloo, frontend and mock-sso all working together such that you can navigate the frontend, be correctly authenticated and see expected data.

### Checklist
